src/compas/topology/connectivity.py

```diff
@@ -41,6 +45,53 @@ def connectivity_from_edges(edges):
     """"""
     raise NotImplementedError
 
+def join_lines_to_polylines(lines):
+    """Join polylines from lines. The polylines stop at points connectng more than two lines.
+
+    Parameters
+    ----------
+    lines : list
+        List of lines as tuples of their extremity coordinates.
+
+    Returns
+    -------
+    polylines: list
+        The polylines. If the polyline is closed, the two extremities are the same.
+
+    """
+
+    # create graph from line extremities
+    network = Network.from_lines([(line[0], line[-1]) for line in lines])
+
+    polylines = []
+    edges_to_visit = list(network.edges())
```

Comment:
Since you're doing frequent lookups on `edges_to_visit` inside the loop, I suspect it might be slow-ish for networks with lots of edges. Maybe you could change this to use a dictionary instead?

Reply:
good point! I switched to a set instead of a list.

Reply:
👍